File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyiddatasalesorders.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCompaniesCompanyIdDataSalesOrdersPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;
}


export class GetCompaniesCompanyIdDataSalesOrdersQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=orderBy" })
  orderBy?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=pageSize" })
  pageSize?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=query" })
  query?: string;
}


export class GetCompaniesCompanyIdDataSalesOrdersSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeApiKeyAuth;
}


export class GetCompaniesCompanyIdDataSalesOrdersRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdDataSalesOrdersPathParams;

  @SpeakeasyMetadata()
  queryParams: GetCompaniesCompanyIdDataSalesOrdersQueryParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdDataSalesOrdersSecurity;
}


export class GetCompaniesCompanyIdDataSalesOrdersResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatDataContractsDatasetsSalesOrderPagedResponseModel?: shared.CodatDataContractsDatasetsSalesOrderPagedResponseModel;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-client-sdk/pkg/models/operations/getcompaniescompanyiddatasalesorders.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetCompaniesCompanyIDDataSalesOrdersPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
}

type GetCompaniesCompanyIDDataSalesOrdersQueryParams struct {
	OrderBy  *string `queryParam:"style=form,explode=true,name=orderBy"`
	Page     int     `queryParam:"style=form,explode=true,name=page"`
	PageSize *int    `queryParam:"style=form,explode=true,name=pageSize"`
	Query    *string `queryParam:"style=form,explode=true,name=query"`
}

type GetCompaniesCompanyIDDataSalesOrdersSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDDataSalesOrdersRequest struct {
	PathParams  GetCompaniesCompanyIDDataSalesOrdersPathParams
	QueryParams GetCompaniesCompanyIDDataSalesOrdersQueryParams
	Security    GetCompaniesCompanyIDDataSalesOrdersSecurity
}

type GetCompaniesCompanyIDDataSalesOrdersResponse struct {
	CodatDataContractsDatasetsSalesOrderPagedResponseModel *shared.CodatDataContractsDatasetsSalesOrderPagedResponseModel
	ContentType                                            string
	StatusCode                                             int
}
